### PR TITLE
Gitbook rendering updates

### PR DESCRIPTION
**Pull request by jacogr**
Ok, @chevdor is going to kill me...

- First off, this Closes https://github.com/polkadot-js/api/issues/158
- API documentation for classes now are rendering correctly
- Removed the type-* from the doc build with a .nodoc file

Now for the long version -

- Had to move all the docs/*.adoc to docs/*.md - gitbook does not include linked *.md files when using the adoc plugin, so all links to the api, api-provider, etc. *.md files were broken
- Even at this point, the nested SUMMARY is an issue, so had to manually add the Api, {Ws,Http}Provider and ApiRx classes to get them to link

What this does not do -

- No fix for #154 atm, it still shows up incorrectly

### CODE REVIEW

File: typedoc.js [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
-  stripInternal: 'true',
-  theme: 'markdown',
-  ignoreCompilerErrors: true,
   moduleResolution: 'node',
-  suppressExcessPropertyErrors: true,
-  suppressImplicitAnyIndexErrors: true
-}
+  mdEngine: 'gitbook',
+  stripInternal: 'true',
+  theme: 'markdown'
```
Comment (amaurymartiny) [inline typedoc.js]: In package.json, you pass `--theme default --out docs/html`, which overwrites options here.

Comment (jacogr) [reply typedoc.js]: Ahh, package.json is actually not used. (But I didn't want to remove it ... yet, it helps when you can actually test HTML from typedoc as well) 

This is a whole sprawling thing, effectively what is actually called is the build-docs script, which is shared - https://github.com/polkadot-js/dev/blob/master/packages/dev/scripts/polkadot-dev-build-docs.sh

(Hence build calling build-ts & build-docs)

Comment (amaurymartiny) [reply typedoc.js]: Ah, right, this is the obscure part I missed.

Comment (jacogr) [reply typedoc.js]: Obscure is the right word, yes.

**Pull request merged**